### ISSUE TITLE
Some more Pandoc 2 details...

### ISSUE THREAD

**Issue by iandol** [htdebeer/paru#34]
Hi, the first version of the **epic** changelog for Pandoc 2 is now available:

https://github.com/jgm/pandoc/commit/25590b18121bf70340f77bb904958fa12bbba677

 and I saw this:

```
Set `PANDOC_READER_OPTIONS` in environment where filters are run. This contains a JSON representation of `ReaderOptions`, so filters can access it.
```

Worth noting, though doesn't need any explicit paru support I suppose? There is also a detailed section of API changes, which may be useful for paru...

**Comment by htdebeer** [comment]
Thanks for the update!

With respect to the `PANDOC_READER_OPTIONS`: If it is in the environment, anyone can access it from Ruby via `ENV["PANDOC_READER_OPTIONS"]`. Then again, it might be nice to have access to it via a instance method / property on Paru::PandocFilter.

Anyway, given the huge amount of changes, paru probably needs to be updated to accommodate them all. After the release of pandoc2, I intend to move the paru-for-pandoc2 to master and create a 0.3 version of paru that is compatible with pandoc 2. From that version on, paru will also drop support for pandoc 1; support for pandoc1 will stay at version 0.2.x.

Do you have any idea when version 2 of pandoc will be released? Then I can plan this work ahead. Otherwise I'll do it after the release.

**Comment by iandol** [comment]
Hi Huub, I read somewhere the end of October, though there is still one unresolved issue (the div markup one, perhaps will not make it into 2...), and a linux packaging issue.

I haven't yet checked what exactly is in `PANDOC_READER_OPTIONS`, but wrapping it with paru methods/properties is not essential...

I have been using Pandoc 2 with pandocomatic/paru without any issues so I think for end users you have already done an excellent job (I know your testcase results are different hence the 0.3 branch)...

**Comment by htdebeer** [comment]
It is good to hear the pandoc2 branch seems to be working fine. This suggests that pandoc2 is largely backwards compatible with pandoc1. I do wonder about the new features introduced in pandoc2, though. Unless we try them all, we do not know if they work okay. Over-all, this means that we can release paru 0.3 when pandoc2 is released and then start gathering issues and fixing them for the new stuff in pandoc2.

**Comment by iandol** [comment]
Only the Pandoc API docs left:

https://github.com/jgm/pandoc/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22pandoc+2.0%22

The fenced DIV are now implemented and are great!

**Comment by iandol** [comment]
And release: https://github.com/jgm/pandoc/releases/tag/2.0

**Comment by htdebeer** [comment]
Thanks for the notice. I'll release paru 0.3 somewhen this week and investigate the work to be done to become completely pandoc 2.0 compatible.

**Comment by htdebeer** [comment]
I have released paru-0.3.0.0. It is mostly the same as the version 0.2.x branch, but I have moved the development environment from pandoc1 to pandoc2. 

- I fixed one test to handle fenced DIVs
- I fixed one typo in the list with options
- the fall-back pandoc version is changed from version 1 to 2

Maybe it is just me, but running the tests using pandoc2 does seem to take a lot longer than running them on pandoc1.
